lib/elixir/lib/uri.ex

```diff
@@ -903,7 +903,7 @@ defmodule URI do
   @spec merge(t | binary, t | binary) :: t
   def merge(uri, rel)
 
-  def merge(%URI{host: nil}, _rel) do
+  def merge(%URI{scheme: nil}, _rel) do
     raise ArgumentError, "you must merge onto an absolute URI"
```

Comment:
I am not sure if this message is correct. Also, shouldn't be fine to merge into `//foo/bar`? Should we add a test for that?

Reply:
Regarding the error message: I believe the message has become more accurate with this change. But "base must be an absolute URI" might be clearer.

Regarding `//foo/bar`: Such a network-path reference, is still a relative URI and should IMO lead to this error when used as a base.